src/engine/routing_algorithms/alternative_path_mld.cpp

```diff
@@ -0,0 +1,786 @@
+
+                LevelID sublevel = level - 1;
+
+                // Here heaps can be reused, let's go deeper!
+                forward_heap.Clear();
+                reverse_heap.Clear();
+                forward_heap.Insert(source, 0, {source});
+                reverse_heap.Insert(target, 0, {target});
+
+                // TODO: when structured bindings will be allowed change to
+                // auto [subpath_weight, subpath_source, subpath_target, subpath] = ...
+                EdgeWeight subpath_weight;
+                std::vector<NodeID> subpath_nodes;
+                std::vector<EdgeID> subpath_edges;
+                std::tie(subpath_weight, subpath_nodes, subpath_edges) = search(search_engine_data,
+                                                                                facade,
+                                                                                forward_heap,
+                                                                                reverse_heap,
+                                                                                force_loop_forward,
+                                                                                force_loop_backward,
+                                                                                INVALID_EDGE_WEIGHT,
+                                                                                sublevel,
+                                                                                parent_cell_id);
+                BOOST_ASSERT(!subpath_edges.empty());
+                BOOST_ASSERT(subpath_nodes.size() > 1);
+                BOOST_ASSERT(subpath_nodes.front() == source);
+                BOOST_ASSERT(subpath_nodes.back() == target);
+                unpacked_nodes.insert(
+                    unpacked_nodes.end(), std::next(subpath_nodes.begin()), subpath_nodes.end());
+                unpacked_edges.insert(
+                    unpacked_edges.end(), subpath_edges.begin(), subpath_edges.end());
+            }
+        }
+
+        WeightedViaNodeUnpackedPath unpacked_path{
+            WeightedViaNode{packed_path_via, packed_path_weight},
+            std::move(unpacked_nodes),
+            std::move(unpacked_edges)};
+
+        unpacked_paths.push_back(std::move(unpacked_path));
+    }
+
+    //
+    // Filter and rank a second time. This time instead of being fast and doing
+    // heuristics on the packed path only we now have the detailed unpacked path.
+    //
+
+    auto unpacked_paths_last = end(unpacked_paths);
+
+    for (std::size_t i = 1; i < unpacked_paths.size(); ++i)
+    {
+        for (std::size_t j = 0; j < i; ++j)
+        {
+            unpacked_paths_last = filterUnpackedPathsBySharing(
+                unpacked_paths[i], begin(unpacked_paths) + i, unpacked_paths_last);
+        }
+    }
+
+    // Todo: rank a last time by weight and sharing here, then just take first n
```

Comment:
what about stretch, which depends on the sharing (as outlined above).

Also: alpha/gamma/epsilon should depend on the shape of the detour to allow more tuning.